scripts/cluster_network.py

```diff
@@ -366,8 +367,11 @@ def reduce_network(n, buses):
         return nr
 
     def busmap_for_country(x):
+
+        # A number of the countries in the clustering can be > 1
         if isinstance(n_clusters, pd.Series):
-            n_cluster_c = n_clusters[x.name]
+            # TODO Verify why n_clusters[x.name] is not working properly
+            n_cluster_c = n_clusters[x.country][0]
```

Comment:
Is this change needed?

Reply:
Yes, that's needed because we have switched-on the sub-networks. Have changed subsetting back from `[x.country][0]` to `[x.name[0]]` for better match with the initial implementation but `[x.name]` is still not working. Our implementation differs from PyPSA-Eur one with some details. Probably this needs to be investigated better as a warning `ConvergenceWarning: Number of distinct clusters (...) found smaller than n_clusters (...). Possibly due to duplicate points in X. kmeans.fit(points)` appears for some clusters values if a group of countries is considered.

Reply:
Did you activate subnetworks in this branch? because I have never experienced this issue

Reply:
Yes, it was needed to get proper management of the AC and DC nodes. I have commented this row

https://github.com/pypsa-meets-africa/pypsa-africa/blob/66a162573b2cee9bf0c8622a4fb5e3c15193e70f/scripts/cluster_network.py#L350

and uncommented this one

https://github.com/pypsa-meets-africa/pypsa-africa/blob/66a162573b2cee9bf0c8622a4fb5e3c15193e70f/scripts/cluster_network.py#L396

Reply:
Ok!